### DISCUSSION TITLE
Is there any document about the principle of fltk-rs?

### DISCUSSION THREAD

**Discussion by Lydanne** [fltk-rs/fltk-rs#1244] [Q&A] [answered]
Is there any document about the principle of fltk-rs? I always have some unexpected problems when I do some examples. I want to have a deeper understanding through the principle

**Comment by MoAlyousef** [comment, marked as answer]
There's no such document. There's a [book](https://www.fltk.org/doc-1.4/fltk.pdf) on programming FLTK in C++, and an [mdbook](https://fltk-rs.github.io/fltk-book/) on programming fltk-rs in Rust. Generally speaking, gui programming is difficult. Using a crossplatform gui can be more difficult if specific platform-dependent effects are required (tray icons, setting window levels ...etc), which crossplatform guis usually don't cover in their api.
